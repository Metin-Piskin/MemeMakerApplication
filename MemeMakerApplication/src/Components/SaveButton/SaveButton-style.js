import { StyleSheet } from "react-native";


export default StyleSheet.create({
    save: {
        backgroundColor: '#2C2C2E',
        borderColor: '#fff',
        justifyContent: 'center',
        alignItems: 'center',
        alignSelf: 'center',
        width: 105,
        height: 60,
        borderWidth: 5,
        marginVertical: 5,
        borderRadius: 10,
        flexDirection: 'row'
    },
})