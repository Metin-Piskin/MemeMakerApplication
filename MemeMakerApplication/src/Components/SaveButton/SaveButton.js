import React from 'react';
import { TouchableOpacity } from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';

import styles from './SaveButton-style';

const SaveButton = ({ onPress, iconName }) => {
    return (
        <TouchableOpacity
            onPress={onPress}
            testID='save-button-press'
            style={styles.save}
        >
            <MaterialIcons
                name={iconName}
                size={40}
                color='#fff'
            />
        </TouchableOpacity>
    )
}

export default SaveButton;