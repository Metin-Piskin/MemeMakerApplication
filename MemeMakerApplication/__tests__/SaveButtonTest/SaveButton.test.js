import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import SaveButton from '../../src/Components/SaveButton';
import SaveButtonStyles from '../../src/Components/SaveButton/SaveButton-style';

jest.mock('react-native-vector-icons/MaterialIcons', () => 'MaterialIcons');

test('renders correctly', () => {
    const comp = render(<SaveButton />);
    expect(comp).toMatchSnapshot();
});

test('should render press correctly', () => {
    const mockFunction = jest.fn();
    const comp = render(<SaveButton onPress={mockFunction} />);

    const SaveButtonPress = comp.getByTestId('save-button-press');
    fireEvent(SaveButtonPress, 'press');

    expect(mockFunction).toBeCalled();
});

test('should render default theme style', () => {
    const comp = render(<SaveButton />);
    const SaveButtonContainerStyle = comp.getByTestId('save-button-press').props.style;

    expect(SaveButtonContainerStyle).toMatchObject(SaveButtonStyles.save);
});