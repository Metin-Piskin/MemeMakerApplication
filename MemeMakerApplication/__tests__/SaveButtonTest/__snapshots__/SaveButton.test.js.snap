// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#2C2C2E",
      "borderColor": "#fff",
      "borderRadius": 10,
      "borderWidth": 5,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "marginVertical": 5,
      "opacity": 1,
      "width": 105,
    }
  }
  testID="save-button-press"
>
  <MaterialIcons
    color="#fff"
    size={40}
  />
</View>
`;
